src/execution/operator/aggregate/physical_window.cpp

```diff
@@ -663,8 +663,12 @@ static void ComputeWindowExpression(BoundWindowExpression *wexpr, ChunkCollectio
 
 		switch (wexpr->type) {
 		case ExpressionType::WINDOW_AGGREGATE: {
-			res = segment_tree->Compute(bounds.window_start, bounds.window_end);
-			break;
+			const auto &source = segment_tree->Compute(bounds.window_start, bounds.window_end);
+			auto &target_chunk = output.GetChunkForRow(row_idx);
+			auto &target = target_chunk.data[output_col];
+			const auto target_offset = row_idx % STANDARD_VECTOR_SIZE;
+			VectorOperations::Copy(source, target, 1, 0, target_offset);
```

Comment:
Perhaps we could let the `aggregate.finalize` take in a `result_offset` parameter. That way we can avoid the Copy entirely, as finalize can write the result directly into the vector at the correct position. That would also avoid the constant vector problems you mentioned.

Reply:
We can do that, but every `Finalize` method in the code will have to change.

Reply:
I think that should actually not be too difficult `AggregateExecutor::Finalize`:

```cpp
        // AggregateExecutor::Finalize
	template <class STATE_TYPE, class RESULT_TYPE, class OP>
	static void Finalize(Vector &states, FunctionData *bind_data, Vector &result, idx_t count) {
		if (states.GetVectorType() == VectorType::CONSTANT_VECTOR) {
			result.SetVectorType(VectorType::CONSTANT_VECTOR);

			auto sdata = ConstantVector::GetData<STATE_TYPE *>(states);
			auto rdata = ConstantVector::GetData<RESULT_TYPE>(result);
			OP::template Finalize<RESULT_TYPE, STATE_TYPE>(result, bind_data, *sdata, rdata,
			                                               ConstantVector::Validity(result), 0);
		} else {
			D_ASSERT(states.GetVectorType() == VectorType::FLAT_VECTOR);
			result.SetVectorType(VectorType::FLAT_VECTOR);

			auto sdata = FlatVector::GetData<STATE_TYPE *>(states);
			auto rdata = FlatVector::GetData<RESULT_TYPE>(result);
			for (idx_t i = 0; i < count; i++) {
				OP::template Finalize<RESULT_TYPE, STATE_TYPE>(result, bind_data, sdata[i], rdata,
				                                               FlatVector::Validity(result), i);
			}
		}
	}

```

And internally use a method such as this:

```cpp
	template <class T, class STATE>
	static void Finalize(Vector &result, FunctionData *, STATE *state, T *target, ValidityMask &mask, idx_t idx) {
		if (!state->isset) {
			mask.SetInvalid(idx);
		} else {
			target[idx] = state->value;
		}
	}
```

So we would only need to change AggregateExecutor::Finalize together with the (small number of) aggregates that manually implement the Finalize method. 

Anyway, it could also be done in a subsequent PR if you prefer.